src/pages/en/core-concepts/routing.md

```diff
@@ -34,119 +34,81 @@ Astro uses standard HTML [`<a>` elements](https://developer.mozilla.org/en-US/do
+A single Astro page file can specify dynamic route parameters in its filename to generate matching pages. For example, you might create an `authors/[author].astro` file that generates a bio page for every author on your blog. `author` becomes a _parameter_ that you can access from inside the page.
 
-:::note
-Even dynamically-created pages and routes are generated at build time.
-:::
-
-Astro pages that create dynamic routes must:
-
-1. use `[bracket]` notation to identify the dynamic parameters
-
-2. export a `getStaticPaths()` function to specify exactly which paths will be pre-rendered by Astro.
+In Astro's default static output mode, these pages are generated at build time, and so you must predetermine the list of `author`s that get a corresponding file. In SSR mode, a page will be generated on request for any route that matches.
 
+### Static (SSG) Mode
 
-### Named Parameters
+Because all routes must be determined at build time, a dynamic route must export a `getStaticPaths()` that returns an array of objects with a `params` property. Each of these objects will generate a corresponding route.
 
-You can generate routes with a `[named]` parameter by providing your `getStaticPaths()` function the values to use like so:
+`[dog].astro` defines the dynamic `dog` parameter in its filename, so the objects returned by `getStaticPaths()` must include `dog` in their `params`. The page can then access this parameter using `Astro.params`.
 
-```astro
+```astro title="src/pages/dogs/[dog].astro"
 ---
-// src/pages/dogs/[dog].astro
-
 export function getStaticPaths() {
   return [
-    // Generates: /dogs/clifford
     {params: {dog: 'clifford'}},
-    // Generates: /dogs/rover
     {params: {dog: 'rover'}},
-    // Generates: /dogs/spot
     {params: {dog: 'spot'}},
   ];
 }
+
+const { dog } = Astro.params;
 ---
+<div>Good dog, {dog}!</div>
 ```
 
-📚 Learn more about [`getStaticPaths()`](/en/reference/api-reference/#getstaticpaths).
-
-Routes can be generated from multiple named parameters, at any level of the filepath:
+This will generate three pages: `dogs/clifford`, `dogs/rover`, and `dogs/spot`, each displaying the corresponding dog name.
 
-- `pages/blog/[slug].astro` → (`/blog/hello-world`, `/blog/post-2`, etc.)
-- `pages/[username]/settings.astro` → (`/fred/settings`, `/drew/settings`, etc.)
-- `pages/[lang]-[version]/info.astro` → (`/en-v1/info`, `/fr-v2/info`, etc.)
+The filename can include multiple parameters, which must all be included in the `params` objects in `getStaticPaths()`:
 
-#### The `Astro.params` object
-
-Astro components that generate routes dynamically have access to an `Astro.params` object for each route. This allows you to use those generated parts of the URL in your component script and template.
-
-```astro / (id) |{id}/ /(?<!//.*)Astro.params/
+```astro title="src/pages/[lang]-[version]/info.astro"
```

Comment:
Ooh! I'd always assumed these had to be separated by slashes. Cool that you can do this too!